### PR TITLE
Fixes:[T338267] Use SMTP to send Emails to the users

### PR DESCRIPTION
**Pull request by okerekechinweotito**
Fixes: [T338267](https://phabricator.wikimedia.org/T338267)

## Proposed Changes
- Add an optional Email Input for status update
- Send an Email to the user after the Job resolves with the job status - either completed or failed

NOTE - According to the [docs](https://wikitech.wikimedia.org/wiki/Help:Toolforge/Email)  The senders address or TOOL_EMAIL should be in this format toolname.anything@toolforge.org. In our case should be bub2@toolforge.org

## Files Created/Updated

- bull/email-queue/consumer.js - configure the Toolforge SMTP for outgoing mail 
- components/Books.js , components/InputEmail.js - Add InputEmail component
- .env.example - pass in description for TOOL_FORGE_EMAIL and TOOL_FORGE_PASSWORD env values
- README.md - update readme.md with description on how to get ToolForge credentials
- bull/google-books-queue/consumer.js - uncomment EmailProducer and pass in email data
- bull/pdl-queue/consumer.js - uncomment EmailProducer and pass in email data
- bull/trove-queue/consumer.js - uncomment EmailProducer and pass in email data

## Checklist 
- [x] Coding Conventions are followed.
- [x] Comments are used for Documenting the Code.
- [x] Correct File Names are mentioned.

### CODE REVIEW

**Comment by coderwassananmol** [comment]
@okerekechinweotito Thanks for the PR. :)
However, we found a better (preferred) way to send Email from a tool. Please look at the documentation [here](https://wikitech.wikimedia.org/wiki/Help:Toolforge/Email#Sending_via_SMTP). I will also update the description of the Phabricator task.
Sorry for making it a double effort!

**Comment by okerekechinweotito** [comment]
@coderwassananmol 
I have pushed new changes to this PR and also updated the PR description.
Please review

**Comment by coderwassananmol** [comment]
@okerekechinweotito Please add the option to enter email while uploading the book. Keep it optional.

**Comment by okerekechinweotito** [comment]
@coderwassananmol 
I have added an optional Email Input

Screenshot (33)

**Comment by okerekechinweotito** [comment]
@Parthiv-M 
I have done the requested changes

File: bull/email-queue/consumer.js [hunk 1]
```diff
@@ -20,10 +22,10 @@ EmailQueue.on("active", (job, jobPromise) => {});
 EmailQueue.on("completed", (job, result) => {});
 
 EmailQueue.process((job, done) => {
-  if (job.data.email != "") {
+  if (job.data.userName != "") {
     const mailOptions = {
-      from: process.env.email, // sender address
-      to: job.data.email, // list of receivers
+      from: process.env.TOOL_FORGE_EMAIL, // sender address
+      to: `${job.data.userName}@toolforge.org`, // list of receivers
```
Comment (coderwassananmol) [inline bull/email-queue/consumer.js:28]: @okerekechinweotito Why are we changing the "to" email address to the toolforge.org domain? Shouldn't this be the email address that is collected during the upload? I'm guessing not everyone uploading will have a Wikitech account.

File: components/InputEmail.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    <div>
+      <h4>3. Enter Email Address</h4>
+      <div className="cdx-text-input input-group">
+        <input
+          className="cdx-text-input__input"
+          id="email"
+          name="email"
+          type="email"
+          disabled={inputDisabled}
+          placeholder="user@gmail.com"
```
Comment (coderwassananmol) [inline components/InputEmail.js:12]: Please change this to: "Enter e-mail address (optional)"

**Review by coderwassananmol — CHANGES_REQUESTED** [review]

File: components/InputEmail.js (hunk 2 above)
Comment (okerekechinweotito) [reply components/InputEmail.js]: @coderwassananmol 
fixed

File: bull/email-queue/consumer.js (hunk 1 above)
Comment (okerekechinweotito) [reply bull/email-queue/consumer.js]: @coderwassananmol 
I have made the requested changes

File: bull/email-queue/consumer.js [hunk 3]
```diff
@@ -8,10 +8,12 @@ const logger = winston.loggers.get("defaultLogger");
...
 const transporter = nodemailer.createTransport({
-  service: process.env.service,
+  host: "mail.tools.wmcloud.org",
+  port: 465, // set PORT to 587 if secure:false for TSL
+  secure: true,
   auth: {
-    user: process.env.email,
-    pass: process.env.password,
+    user: process.env.TOOL_FORGE_EMAIL,
+    pass: process.env.TOOL_FORGE_PASSWORD,
```
Comment (coderwassananmol) [inline bull/email-queue/consumer.js:16]: Please remove "," at the end.

File: bull/email-queue/consumer.js [hunk 4]
```diff
@@ -8,10 +8,12 @@ const logger = winston.loggers.get("defaultLogger");
...
   },
```
Comment (coderwassananmol) [inline bull/email-queue/consumer.js:17]: Please remove "," at the end.

**Comment by okerekechinweotito** [comment]
@coderwassananmol  fixed

**Comment by coderwassananmol** [comment]
@okerekechinweotito I had a larger discussion with the Wikimedia tech team and they recommended using `API:EmailUser` to send the emails as the SMTP approach may involve spamming someone else's email because we are capturing email ID separately. We can revert back to the previous commit changes and while the user is authorizing with their Wikimedia account, we collect their email ID (I think we're already doing that), we send an email to them. To send an email, we will collect their consent while uploading with a checkbox that should say, "Notify updates via e-mail". Please add an info icon next to it which should say: "BUB2 will send an email to your email ID associated with your Wikimedia account regarding the success or failure of the upload. If no email is added to the account, email will not be sent."

@okerekechinweotito You may choose to close this PR and submit a fresh PR if you like.

**Comment by okerekechinweotito** [comment]
Closed as Duplicate. 
Replacement PR available at - https://github.com/coderwassananmol/BUB2/pull/227

**Pull request closed without merging**